### PR TITLE
feat(core): add `invalidateOnUnmount` and `invalidateOnClose` prop on `autoSave` feature

### PR DESCRIPTION
**Pull request by yildirayunlu**
`autoSave` mutation not invalide when it was successful. Now `list` and `many` queries will be invalidated by default. 

Also added `invalidateOnUnmountDetailCache` prop to work when `unmount`. This prop will invalidate the `detail` query when `true` is set.

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: packages/core/src/hooks/form/useForm.ts [hunk 1]
```diff
@@ -460,6 +460,11 @@ export const useForm = <
     ): Promise<UpdateResponse<TResponse> | void> | void => {
         if (!resource || !isEdit) return;
 
+        const defaultInvalidates: (keyof IQueryKeys)[] = ["list", "many"];
+        if (autoSave?.invalidateOnUnmountDetailCache) {
+            defaultInvalidates.push("detail");
+        }
```
Comment (aliemir) [inline packages/core/src/hooks/form/useForm.ts:466]: I'm not sure if I can follow what's happening here. I can't see a relation with unmount, how do these lines work? 🤔

File: documentation/docs/api-reference/mantine/hooks/form/useModalForm.md [hunk 2]
```diff
@@ -919,6 +921,22 @@ useModalForm({
     }
 })
 ```
+
+#### `invalidateOnUnmountDetailCache`
+
+If you want to invalidate the `detail` query cache when the modal is closed, you can set the `invalidateOnUnmountDetailCache` parameter to `true`. Default value is `false`.
```
Comment (alicanerdurmaz) [inline documentation/docs/api-reference/mantine/hooks/form/useModalForm.md:927]: I guess this won't work when the modal is closed. In our examples, `useModalForm` is still mounted

File: documentation/docs/api-reference/mantine/hooks/form/useForm.md [hunk 3]
```diff
@@ -1003,6 +1005,22 @@ useForm({
 });
 ```
 
+#### `invalidateOnUnmountDetailCache`
+
+If you want to invalidate the `detail` query cache when the form is unmounted, you can set the `invalidateOnUnmountDetailCache` parameter to `true`. Default value is `false`.
```
Comment (alicanerdurmaz) [inline documentation/docs/api-reference/mantine/hooks/form/useForm.md:1010]:
```suggestion
If you want to invalidate the `detail` query cache when the `useForm` is unmounted, you can set the `invalidateOnUnmountDetailCache` parameter to `true`. Default value is `false`.
```

**Review by alicanerdurmaz — CHANGES_REQUESTED** [review]

File: documentation/docs/api-reference/mantine/hooks/form/useModalForm.md (hunk 2 above)
Comment (yildirayunlu) [reply documentation/docs/api-reference/mantine/hooks/form/useModalForm.md:927]: You are right, here I can update it as follows; `when the `useModalForm` is unmounted`

File: packages/mantine/src/hooks/form/useModalForm/index.ts [hunk 4]
```diff
@@ -246,7 +253,17 @@ export const useModalForm = <
     };
 
     const { warnWhen, setWarnWhen } = useWarnAboutChange();
+    const {} = useMantineFormResult;
```
Comment (salihozdemir) [inline packages/mantine/src/hooks/form/useModalForm/index.ts:256]:
```suggestion
```

File: documentation/docs/api-reference/antd/hooks/form/useForm.md [hunk 5]
```diff
@@ -897,6 +899,19 @@ useForm({
     },
 });
 ```
+#### `invalidateOnUnmount`
+
+If you want to invalidate the `list`, `many` and `detail` queries from the current resource when auto save mutation is successful, you can use the `invalidateOnUnmount` prop. This prop is useful when you want to invalidate the `list`, `many` and `detail` queries from the current resource. However, you can use the `invalidates` prop if you want to customize it.
```
Comment (salihozdemir) [inline documentation/docs/api-reference/antd/hooks/form/useForm.md:904]: Here the before, we should say something like following:

By default the `autoSave` feature does not invalidate queries. If you want to invalidate, you can use the `invalidateOnUnmount` prop.

**Review by salihozdemir — CHANGES_REQUESTED** [review]

**Review by alicanerdurmaz — APPROVED** [review]

**Review by omeraplak — APPROVED** [review]

File: .changeset/fast-dodos-think.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+---
+"@refinedev/core": patch
+---
+
+feat: `autoSave` mutation not invalide when it mutation successful. Now `list`,`many` and `detail` queries will be invalidated by default. Also added [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to work when `unmount`. However, you can use the `invalidates` prop if you want to customize it.
```
Comment (BatuhanW) [inline .changeset/fast-dodos-think.md:5]:
```suggestion
feat: `autoSave` mutation wasn't being invalidated when the mutation is successful. Now `list`,`many` and `detail` queries will be invalidated by default. Also added [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to to trigger invalidate on `unmount`. You can use `invalidates` prop to customize it.
```

File: .changeset/six-parrots-joke.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
+---
+"@refinedev/antd": patch
+---
+
+fix: when using [`useForm`](https://refine.dev/docs/api-reference/antd/hooks/form/useForm/), `autoSave` parameters are passed to `@refinedev/core/useForm` hook.
```
Comment (BatuhanW) [inline .changeset/six-parrots-joke.md:5]:
```suggestion
fix: [`useForm`](https://refine.dev/docs/api-reference/antd/hooks/form/useForm/)'s, `autoSave` parameters not passed to `@refinedev/core/useForm` hook.
```

**Review by BatuhanW — CHANGES_REQUESTED** [review]

File: .changeset/clever-elephants-decide.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+---
+"@refinedev/react-hook-form": patch
+---
+
+feat: add `invalidateOnUnmount` prop to [`useForm`](https://refine.dev/docs/packages/documentation/react-hook-form/useForm/) hook.
+feat: add `invalidateOnUnmount` and `invalidateOnClose` prop to [`useModalForm`](https://refine.dev/docs/packages/documentation/react-hook-form/useModalForm/) hook.
+From now, you can use the use this props to invalidate queries upon unmount or close.
```
Comment (BatuhanW) [inline .changeset/clever-elephants-decide.md:7]:
```suggestion
From now on, you can use these props to invalidate queries upon unmount and/or close.
```

File: .changeset/fast-dodos-think.md [hunk 9]
```diff
@@ -0,0 +1,6 @@
+---
+"@refinedev/core": patch
+---
+
+feat: add [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to [`useForm`](https://refine.dev/docs/api-reference/core/hooks/useForm) hook.
+From now, you can use the [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to invalidate queries upon unmount.
```
Comment (BatuhanW) [inline .changeset/fast-dodos-think.md:6]:
```suggestion
From now on, you can use the [`invalidateOnUnmount`](https://refine.dev/docs/api-reference/core/hooks/useForm/#invalidateonunmount) prop to invalidate queries upon unmount.
```

File: documentation/docs/api-reference/antd/hooks/form/useDrawerForm.md [hunk 10]
```diff
@@ -395,6 +403,38 @@ useDrawerForm({
 });
 ```
 
+#### `invalidateOnUnmount`
+
+> Default: `false`
+
+This prop is useful when you want to invalidate the `list`, `many` and `detail` queries from the current resource when the hook is unmounted. By default, it invalidates the `list`, `many` and `detail` queries associated with the current resource. Also, You can use the `invalidates` prop to select which queries to invalidate.
```
Comment (BatuhanW) [inline documentation/docs/api-reference/antd/hooks/form/useDrawerForm.md:410]: Can we also add example about how to use `invalidates` prop if we don't have already?

**Review by BatuhanW — APPROVED** [review]
LGTM 👍🏼 Added minor comments.

**Pull request merged**